compiler/src/main/kotlin/io/spine/protodata/AstExtension.kt

```diff
@@ -93,6 +100,79 @@ public fun Field.isRepeated(): Boolean = isMap() || isList()
  */
 public fun Field.isPartOfOneof(): Boolean = hasOneofName()
 
+/**
+ * Looks up an option value by the [optionName].
+ *
+ * @return the value of the option or a `null` if the option is not found
+ */
+public fun <T : Message> Iterable<Option>.find(optionName: String, cls: Class<T>): T? {
+    val value = firstOrNull { it.name == optionName }?.value
+    return value?.unpack(cls)
+}
+
+/**
+ * Obtains the path to the `.java` file, generated from this message.
+ *
+ * The class which represents this message might not be the top level class of the Java file.
+ */
+public fun MessageType.javaFile(declaredIn: File): Path {
+    val packageName = declaredIn.javaPackage()
+    val javaMultipleFiles = declaredIn.javaMultipleFiles()
+    val topLevelClassName = if (javaMultipleFiles) {
+        name.simpleName
+    } else {
+        declaredIn.javaOuterClassName()
+    }
+    return Paths.get(packageName.replace('.', '/'), "$topLevelClassName.java")
+}
+
+/**
+ * Obtains the full name of the Java class, generated from this message.
+ */
+public fun MessageType.javaClassName(declaredIn: File): String {
```

Comment:
This method isn't used either.

I imagine it may be very well required in future, but let's cover it with some tests in scope of this PR.